### ISSUE TITLE
Replace legacy "coloring" in phase-field/Marmot

### ISSUE THREAD

**Issue by permcody** [idaholab/moose#8833]
### Description of the enhancement or error report

After #8805 is merged, we'll want to deprecate and remove the legacy coloring option in the phase-field module and derived applications.

### Rationale for the enhancement or information for reproducing the error

The legacy option may produce invalid colorings. It really needs to go.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

Existing tests and example inputs may need modifications and regolding.

**Comment by permcody** [comment]
Tag @dschwen - Here's a new ticket to get rid of that legacy coloring option.

**Comment by permcody** [comment]
All legacy options have been removed from phase-field and Marmot.
